src/OpenTelemetry.Contrib.Instrumentation.Wcf/TelemetryClientMessageInspector.cs

```diff
@@ -136,9 +136,13 @@ public void AfterReceiveReply(ref Message reply, object correlationState)
 
             state.SuppressionScope?.Dispose();
 
-            Activity activity = state.Activity;
-            if (activity != null)
+            if (state.Activity is Activity activity)
             {
+                if (Activity.Current != activity)
+                {
+                    Activity.Current = activity;
```

Comment:
why do we have to set Current here? what would happen to the original Current, if it was not null?

Reply:
I'm going to reply to this on the comment above.